### ISSUE TITLE
Issue running on iOS

### ISSUE THREAD

**Issue by se1exin** [jzharris/Unity-MNIST#3]
Attempting to run the example on iOS throws the following error (using the frozen graphs from the repo):

```
Failed: No OpKernel was registered to support Op 'Switch' with these attrs.  Registered devices: [CPU], Registered kernels:
  device='GPU'; T in [DT_FLOAT]
  device='GPU'; T in [DT_INT32]
  device='GPU'; T in [DT_BOOL]
  device='GPU'; T in [DT_STRING]
  device='CPU'; T in [DT_INT32]
  device='CPU'; T in [DT_FLOAT]

	 [[Node: dropout_1/cond/Switch = Switch[T=DT_BOOL](dropout_1/keras_learning_phase, dropout_1/keras_learning_phase)]]
```

Which is thrown on the line:
```
 float[,] recurrent_tensor = runner.Run()[0].GetValue() as float[,];
```

Looking deeper, the linked ml-agents plugin uses TF v 1.1.0, however I'm assuming the model was trained with 1.4.

Any ideas on how to either upgrade the unity TF lib, or otherwise fix the issue above?

**Comment by jzharris** [comment]
There are other iOS-TF apps having this issue. The `Switch` kernel is not available in the `DT_BOOL` type for iOS. I did not realize this. I am removing iOS support until I can resolve this issue.

Please try the following solution, which will register the correct data types to `Switch`, along with other kernels lacking datatypes in iOS: https://github.com/tensorflow/tensorflow/issues/9476#issuecomment-313327823

**Comment by se1exin** [comment]
Thanks @jzharris  

We got around the problem by upgrading the TFSharp unity plugin to 0.4 - which uses TF 1.8. We did however need to re-train our models in TF 1.8 as well - but atleast they work now.

Hopefully your findings help anyone else that stumbles across this issue.

Also for reference, the TFSharp Unity plugins are quite hard to find, so here are some links:

v0.4 (TF1.8) - https://s3.amazonaws.com/unity-ml-agents/0.4/TFSharpPlugin.unitypackage

v0.5 (TF1.8) - https://s3.amazonaws.com/unity-ml-agents/0.5/TFSharpPlugin.unitypackage
